src/providers/patreon.js

```diff
@@ -0,0 +1,27 @@
+/** @type {import(".").OAuthProvider} */
+export default function Patreon(options) {
+    return {
+        id: 'patreon',
+        name: 'Patreon',
+        type: 'oauth',
+        version: '2.0',
+        authorization: { 
+          url : 'https://www.patreon.com/oauth2/authorize?response_type=code',
+          params: { scope: 'identity identity[email] identity.memberships', grant_type: 'authorization_code' }
+        },
+        token : {
+          url : 'https://www.patreon.com/api/oauth2/token',
+        },
+        userinfo : {
+          url: 'https://www.patreon.com/api/oauth2/api/current_user'
+        },      
```

Comment:
```suggestion
        token: "https://www.patreon.com/api/oauth2/token",
        userinfo: "https://www.patreon.com/api/oauth2/api/current_user",
```

Just some simplification.